loopy/codegen/instruction.py

```diff
@@ -29,24 +29,29 @@
 from loopy.codegen import Unvectorizable
 from loopy.codegen.result import CodeGenerationResult
 from pymbolic.mapper.stringifier import PREC_NONE
+from pytools import memoize_method
+
+
+@memoize_method
+def _get_new_implemented_domain(chk_domain, implemented_domain):
+
+    chk_domain, implemented_domain = isl.align_two(
+            chk_domain, implemented_domain)
+    chk_domain = chk_domain.gist(implemented_domain)
+
+    new_implemented_domain = implemented_domain & chk_domain
```

Comment:
This confuses me as (IIUC)
1. `gist` performs an intersection while being careful of the dim names, so not sure if `align_two` is of help here, and,
2. why would `chk_domain` and `new_implemented_domain` be any different?

Reply:
I've no clue what any of this does. :smile:

Reply:
@kaushikcfd, what makes you say that `gist`, unlike the rest of ISL, matches dims by name?